src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_manager.h

```diff
@@ -117,27 +114,30 @@ extern const char* upgrade_error_codes[];
 /**
  * Start listening loop, exits only on error 
  * @param timeout_sec timeout in seconds
+ * @param manager_configs manager configuration parameters
  * @return only on errors, socket will be closed
  * */
-void wm_agent_upgrade_listen_messages(int timeout_sec);
+void wm_agent_upgrade_listen_messages(int timeout_sec, wm_manager_configs manager_configs);
 
 /**
  * Process and upgrade command. Create the task for each agent_id, dispatches to task manager and
  * then starts upgrading process.
  * @param agent_ids array with the list of agents id
  * @param task pointer to a wm_upgrade_task structure
+ * @param manager_configs manager configuration parameters
  * @return string with the response
  * */
-char* wm_agent_upgrade_process_upgrade_command(const int* agent_ids, wm_upgrade_task* task);
+char* wm_agent_upgrade_process_upgrade_command(const int* agent_ids, wm_upgrade_task* task, wm_manager_configs manager_configs);
```

Comment:
Since manager_configs is used as read-only it could be defined as const. It is also better to use a pointer, to avoid creating a copy of the structure.
const wm_manager_configs* manager_configs